src/sentry/services/hybrid_cloud/user.py

```diff
@@ -12,9 +14,33 @@
 from sentry.silo import SiloMode
 
 
+@dataclass(frozen=True)
+class APIUser:
+    id: int = -1
+    name: str = ""
+    email: str = ""
+    username: str = ""
+    actor_id: int = -1
+
+    # TODO: Extract to base dataclass?
+    @classmethod
+    def from_orm(cls, orm: Any):
+        # fetch the constructor's signature
+        cls_fields = {field for field in signature(cls).parameters}
```

Comment:
This totally works, but there's an even more elegant way:

https://docs.python.org/3/library/dataclasses.html#dataclasses.fields

You can even add custom metadata to 'hint' at how unpacking should be done.